### PR TITLE
fix: remove undefined key of stablecoin

### PR DESCRIPTION
**Pull request by ChefJerry**
---

## PR-Codex overview
This PR updates the `MAJOR_DEFINED_STABLE_COINS` object structure and refactors the `useFeeSaved` hook to handle stable coins more efficiently.

### Detailed summary
- Restructured `MAJOR_DEFINED_STABLE_COINS` object with a chain-specific array
- Updated `useFeeSaved` to handle stable coins based on the chain ID
- Improved handling of zero fee scenarios

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/views/Swap/hooks/useFeeSaved.ts [hunk 1]
```diff
@@ -8,20 +8,9 @@ import { useMemo } from 'react'
 const MAJOR_DEFINED_STABLE_COINS = {
```
Comment (chefjackson) [inline apps/web/src/views/Swap/hooks/useFeeSaved.ts:8]: Let's add explicit type definition here. So we can identify the error at build time

Comment (chefjackson) [reply apps/web/src/views/Swap/hooks/useFeeSaved.ts:8]: also can remove the optional mark in line 24

Comment (ChefJerry) [reply apps/web/src/views/Swap/hooks/useFeeSaved.ts]: explicit mark type as `ERC20Token | undefined` and keep optional mark, as `Token[chainId]` will not throw any warning during static check or build time.

File: apps/web/src/views/Swap/hooks/useFeeSaved.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+import { ChainId, Currency, CurrencyAmount, ERC20Token } from '@pancakeswap/sdk'
 import { DAI, USDC, USDT } from '@pancakeswap/tokens'
 import { parseNumberToFraction } from '@pancakeswap/utils/formatFractions'
 import { BigNumber } from 'bignumber.js'
 import { useCurrencyUsdPrice } from 'hooks/useCurrencyUsdPrice'
 import { useMemo } from 'react'
 
-const MAJOR_DEFINED_STABLE_COINS = {
+const MAJOR_DEFINED_STABLE_COINS: {
+  [chainId in ChainId]?: Array<ERC20Token | undefined>
```
Comment (chefjackson) [inline apps/web/src/views/Swap/hooks/useFeeSaved.ts:9]: The list is defined by us right? So we can safely assert it's not undefined here. If any of them is undefined, the error throws on build time

```suggestion
  [chainId in ChainId]?: Array<ERC20Token>
```

File: apps/web/src/views/Swap/hooks/useFeeSaved.ts (hunk 1 above)
Comment (chefjackson) [reply apps/web/src/views/Swap/hooks/useFeeSaved.ts]: hmm ok that's weird. Let's keep it for now. Will double check later on

**Review by chefjackson — APPROVED** [review]

File: apps/web/src/views/Swap/hooks/useFeeSaved.ts (hunk 2 above)
Comment (ChefJerry) [reply apps/web/src/views/Swap/hooks/useFeeSaved.ts:9]: can check this [commit](https://github.com/pancakeswap/pancake-frontend/commit/06276308934e120329aa98e03500d7ad42b50a1d)

build passed with no error

**Pull request merged**